DOC-2321: Add instructions on using filter tabs (#14790)
cockroachdb/docs: StyleGuide.md

The general process to follow and use this is as follows:
  
1. Identify each page to be linked to a filter tab.
    - Make a note of each HTML page name (e.g., `install-cockroachdb-mac.html`).
    - Identify the desired HTML-supported tab name (e.g., `Install <strong>Mac</strong>`).
2. Create an include Markdown file within `_includes/<CRDB version>/filter-tabs` with the following structure:
    ```
    {% assign tab_names_html = "Tab Name 1;Tab Name 2;Tab Name 3" %}
    {% assign html_page_names = "page-name-1.html;page-name-2.html;page-name-3.html" %}

    {% include filter-tabs.md tab_names=tab_names_html page_names=html_page_names page_folder=<CRDB version> %}
    ```
    - `tab_names_html` is a semicolon-separated list of the HTML-supported tab names.
    - `html_page_names` is a semicolon-separated list of the page names with the `.html` extension.
    - `<crdb_version>` is `"cockroachcloud"` (with quotes) for any CockroachDB Cloud docs and `page.version.version` (without quotes) for any versioned docs (v21.2 and later).

> 3. Paste `{% include <CRDB version>/filter-tabs/<filter-tab-include>.html %}` in the desired position on each page listed in `html_page_names`.

Reviewer: Can you clarify what "in the desired position" means? Does it mean that's where the tabs will display?

Author: Yes. How does it look now?

Reviewer: LGTM

```

For more information about the `remote_include` tag, see the README in the [jekyll-remote-include](https://github.com/cockroachdb/jekyll-remote-include) repo.
  
#### Filter tabs
  
On some pages in our docs, there are tabs at the top of the page that will link to different pages at different hyperlinks. For example, in the [Install CockroachDB docs](https://www.cockroachlabs.com/docs/stable/install-cockroachdb.html), there are links to the Mac, Linux, and Windows pages at the top of the page.
  

> You can use [`filter-tabs.md`](https://github.com/cockroachdb/docs/blob/master/_includes/filter-tabs.md) to simplify the process for any `cockroachcloud` docs or docs for v21.2 and later.

Reviewer 2: Based on your phrasing I assume 'the old way' will still work, but its more convenient to use the new way. Be that as it may, it probably makes sense to require the new way, so I recommend:

```suggestion
Use [`filter-tabs.md`](https://github.com/cockroachdb/docs/blob/master/_includes/filter-tabs.md) to specify these tabs for any `cockroachcloud` docs or docs for CockroachDB v21.2 and later.
```

> **Note:** this include file is used only when navigating to different URLs/pages. This should not be used for creating tabs within a single page.

Reviewer 2: ```suggestion
**Note:** this include file only produces tabs that link to different URLs/pages. It cannot be used for creating tabs within a single page.
```

The general process to follow and use this is as follows:
  

> 1. Identify each page to be linked to a filter tab.

Reviewer 2: ```suggestion
1. Identify each page to be linked from a filter tab.
```

>     - Make a note of each HTML page name (e.g., `install-cockroachdb-mac.html`).

Reviewer 2: ```suggestion
    - Make a note of each HTML page filename (e.g., `install-cockroachdb-mac.html`).
```

>     - Identify the desired HTML-supported tab name (e.g., `Install on <strong>Mac</strong>`).

Reviewer 2: ```suggestion
    - Draft a tab name (e.g., `Install on <strong>Mac</strong>`)—the text to display on the tab itself. This supports HTML, not Markdown.
```

>     - Identify the desired HTML-supported tab name (e.g., `Install <strong>Mac</strong>`).

Reviewer 2: ```suggestion
    - Identify the desired HTML-supported tab name (e.g., `Install on <strong>Mac</strong>`).
```

2. Create an include Markdown file within `_includes/<CRDB version>/filter-tabs` with the following structure:
    ```
    {% assign tab_names_html = "Tab Name 1;Tab Name 2;Tab Name 3" %}

>     {% assign html_page_names = "page-name-1.html;page-name-2.html;page-name-3.html" %}

Reviewer 2: For a future improvement, we could rename `html_page_names` to `html_page_filenames`, for clarity.

Author: https://cockroachlabs.atlassian.net/browse/DOC-5506

>     {% include filter-tabs.md tab_names=tab_names_html page_names=html_page_names page_folder=<CRDB version> %}

Reviewer 2: Same here, for a future improvement: `page_names` -> `page_filenames`

    ```
    - `tab_names_html` is a semicolon-separated list of the HTML-supported tab names.

>     - `html_page_names` is a semicolon-separated list of the page names with the `.html` extension.

Reviewer 2: ```suggestion
    - `html_page_names` is a semicolon-separated list of the page filenames with the `.html` extension.
```

    - `<crdb_version>` is `"cockroachcloud"` (with quotes) for any CockroachDB Cloud docs and `page.version.version` (without quotes) for any versioned docs (v21.2 and later).

> 3. Paste `{% include <CRDB version>/filter-tabs/<filter-tab-include>.html %}` in the position you wish to display the tabs in each page listed in `html_page_names`.

Reviewer 2: ```suggestion
3. For each page listed in `html_page_names`, paste `{% include <CRDB version>/filter-tabs/<filter-tab-include>.html %}` in the position where you want the tabs to be included.
```
